org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/ServerRefactoringIssueAcceptor.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2017, 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.ide.server.rename;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.lsp4j.jsonrpc.ResponseErrorException;
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseError;
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseErrorCode;
+import org.eclipse.xtext.ide.refactoring.RefactoringIssueAcceptor;
+import org.eclipse.xtext.util.ITextRegion;
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+import org.eclipse.xtext.xbase.lib.ListExtensions;
+
+/**
+ * @author koehnlein - Initial contribution and API
+ * @since 2.13
+ */
+public class ServerRefactoringIssueAcceptor implements RefactoringIssueAcceptor {
+	public static class Issue {
+		private RefactoringIssueAcceptor.Severity severity;
+
+		private String message;
+	}
+
+	private List<ServerRefactoringIssueAcceptor.Issue> issues = new ArrayList<>();
+
+	@Override
+	public void add(RefactoringIssueAcceptor.Severity severity, String message, URI uri, ResourceSet resourceSet) {
+		addIssue(severity, message);
+	}
+
+	@Override
+	public void add(RefactoringIssueAcceptor.Severity severity, String message, URI resourceUri) {
+		addIssue(severity, message);
+	}
+
+	@Override
+	public void add(RefactoringIssueAcceptor.Severity severity, String message, final EObject element) {
+		addIssue(severity, message);
+	}
+
+	@Override
+	public void add(RefactoringIssueAcceptor.Severity severity, String message, EObject element, ITextRegion region) {
+		addIssue(severity, message);
+	}
+
+	@Override
+	public void add(RefactoringIssueAcceptor.Severity severity, String message, Exception exc, Logger log) {
+		addIssue(severity, message);
+	}
+
+	@Override
+	public void add(RefactoringIssueAcceptor.Severity severity, String message, Object... params) {
+		ServerRefactoringIssueAcceptor.Issue issue = new ServerRefactoringIssueAcceptor.Issue();
+		issue.severity = severity;
+		issue.message = message;
+		issues.add(issue);
+	}
+
+	protected boolean addIssue(RefactoringIssueAcceptor.Severity severity, String message) {
+		ServerRefactoringIssueAcceptor.Issue issue = new ServerRefactoringIssueAcceptor.Issue();
+		issue.severity = severity;
+		issue.message = message;
+		return issues.add(issue);
+	}
+
+	public RefactoringIssueAcceptor.Severity getMaximumSeverity() {
+		if (issues.size() > 0) {
+			ServerRefactoringIssueAcceptor.Issue minBySeverity = IterableExtensions.minBy(issues, (i) -> i.severity);
+			RefactoringIssueAcceptor.Severity severity = null;
```

Comment:
unnecessary local var

Reply:
?

Reply:
since the issues iterable is not empty, minBySeverity cannot be null.
Could be simplified to  `return IterableExtensions.minBy(..).severity`